File: C/ex11.c
#include <stdio.h>
#include <string.h>

int main (void) {
    char age[BUFSIZ],height[BUFSIZ], weight[BUFSIZ];
    printf("How old are you?\n");
    fgets(age, BUFSIZ, stdin); 
    int len_a=strlen(age); //where buff is your char array fgets is using
    if(age[len_a-1]=='\n')
        age[len_a-1]='\0';
    
    printf("How tall are you?\n");
    fgets(height, BUFSIZ, stdin); 
    int len_b=strlen(height); //where buff is your char array fgets is using
    if(height[len_b-1]=='\n')
        height[len_b-1]='\0';

    printf("How much do you weigh?\n");
    fgets(weight, BUFSIZ, stdin); 
    int len=strlen(weight); //where buff is your char array fgets is using
    if(weight[len-1]=='\n')
        weight[len-1]='\0';  

    printf("So, you're %s years old, %s tall, and weigh %s lbs.\n", age, height, weight); 
    return 0; 
}

File: Rust/ex11.rs
use std::io::{self, Write};
use std::fmt::Display;
use std::process;
 
fn main() {
    let age = grab_input("How old are you?")
        .unwrap_or_else(|e| exit_err(&e, e.raw_os_error().unwrap_or(-1)));
    let height = grab_input("How tall are you?")
        .unwrap_or_else(|e| exit_err(&e, e.raw_os_error().unwrap_or(-1)));
    let weight = grab_input("How much do you weigh?")
        .unwrap_or_else(|e| exit_err(&e, e.raw_os_error().unwrap_or(-1)));
    println!("So, you're {} years old, {} tall, and weigh {} lbs.", age, height, weight)
}
 
fn grab_input(msg: &str) -> io::Result<String> {
    let mut buf = String::new();
    print!("{}\n", msg);
    io::stdout().flush()?;
 
    io::stdin().read_line(&mut buf)?;
    Ok(buf.trim().to_string())
}
 
fn exit_err<T: Display>(msg: T, code: i32) -> ! {
    let _ = writeln!(&mut io::stderr(), "Error: {}", msg);
    process::exit(code)
}
